src/tapis-app/Streams/MeasurementsListing/_Layout/Layout.tsx

```diff
@@ -0,0 +1,63 @@
+import { useList } from 'tapis-hooks/streams/measurements';
+import Measurements from '../_components/Measurements';
+import React, { useCallback, useState } from 'react';
+import styles from './Layout.module.scss';
+import { QueryWrapper } from 'tapis-ui/_wrappers';
+
+const Layout: React.FC<{
+  projectId: string;
+  siteId: string;
+  instrumentId: string;
+}> = ({ projectId, siteId, instrumentId }) => {
+  const { data, isLoading, error } = useList({
+    projectId,
+    siteId,
+    instId: instrumentId,
+  });
+
+  const { instrument, site, measurements_in_file, ...measurements } =
+    data?.result ?? {};
+
+  const variables = Object.keys(measurements);
+  const [selected, setSelected] = useState<number>(-1);
+
+  let select = useCallback(
+    (index: number) => {
+      return () => {
+        setSelected(selected === index ? -1 : index);
+      };
+    },
+    //only need to update select function if measurements changed (only time indexes will change)
+    /* eslint-disable-next-line */
+    [measurements]
+  );
+
+  return (
+    <QueryWrapper isLoading={isLoading} error={error}>
+      <div className={styles['variable-list']}>
+        {variables.length ? (
+          variables.map((variable: string, index: number) => {
+            const id = `${index}`;
+            let variableMeasurements = measurements[variable];
+
+            return (
+              <Measurements
+                key={id}
+                id={id}
+                variable={variable}
+                graphWidth={600}
+                measurements={variableMeasurements}
+                select={select(index)}
```

Comment:
This may not be what you want. Usually with a callback wrapped as a closure, it is:

```
select={() => select(index)}
```

Reply:
I don't understand what you mean here. This would not work. The select function returns a callback function bound to the index.

Reply:
The prop select is executed by an onclick function in the Measurment component. If I used the function you proposed, clicking on an element would generate a function and do nothing (unless I generated the function then called it, but this is basically the same as what I already have with extra steps).